pkg/server/server_test.go

```diff
@@ -623,8 +680,122 @@ func TestServerWithFailureScenarios(t *testing.T) {
 	}
 }
 
+func TestSyncUserTxWithServerTLS(t *testing.T) {
```

Comment:
I don't see tests that show what happens when:
1. a client tries to access with http
2. a client tries to access with https his certificate is not from the same CA, etc... 
that is, negative tests

Reply:
Will add such tests